Python/pywarpx/fields.py

```diff
@@ -516,6 +517,12 @@ def __setitem__(self, index, value):
             if not isinstance(ic   , slice) or len(global_shape) < 4: global_shape[3:3] = [1]
             value3d.shape = global_shape
 
+            if libwarpx.libwarpx_so.Config.have_gpu:
```

Comment:
I think only needed for  `ndarray`s:
```suggestion
            if libwarpx.libwarpx_so.Config.have_gpu and isinstance(value, np.ndarray)::
```